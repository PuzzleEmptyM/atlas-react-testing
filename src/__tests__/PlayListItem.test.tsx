import { render } from '@testing-library/react';
import PlayListItem from '../components/PlayListItem';

describe('PlayListItem Component', () => {
  it('renders with required props', () => {
    const { asFragment } = render(
      <PlayListItem title="Test Song" artist="Test Artist" duration="3:45" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with additional prop (className)', () => {
    const { asFragment } = render(
      <PlayListItem
        title="Another Test Song"
        artist="Another Test Artist"
        duration="4:20"
        className="custom-class"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with longer title and artist name', () => {
    const { asFragment } = render(
      <PlayListItem
        title="A Very Long Song Title That Exceeds Normal Length And Should Not Be This Long"
        artist="An Artist With A Very Long Name For Some Reason"
        duration="5:05"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
