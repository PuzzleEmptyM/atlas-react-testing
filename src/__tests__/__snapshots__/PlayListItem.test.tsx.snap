// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem Component > renders with additional prop (className) 1`] = `
<DocumentFragment>
  <div
    class="w-full max-w-lg mt-2.5 custom-class"
  >
    <div
      class="flex justify-between p-2.5"
    >
      <div
        class="text-left"
      >
        <p
          class="text-base font-bold"
        >
          Another Test Song
        </p>
        <p
          class="text-gray-600 font-medium"
        >
          Another Test Artist
        </p>
      </div>
      <p
        class="text-gray-400 font-medium text-lg mt-2"
      >
        4:20
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PlayListItem Component > renders with longer title and artist name 1`] = `
<DocumentFragment>
  <div
    class="w-full max-w-lg mt-2.5 undefined"
  >
    <div
      class="flex justify-between p-2.5"
    >
      <div
        class="text-left"
      >
        <p
          class="text-base font-bold"
        >
          A Very Long Song Title That Exceeds Normal Length And Should Not Be This Long
        </p>
        <p
          class="text-gray-600 font-medium"
        >
          An Artist With A Very Long Name For Some Reason
        </p>
      </div>
      <p
        class="text-gray-400 font-medium text-lg mt-2"
      >
        5:05
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PlayListItem Component > renders with required props 1`] = `
<DocumentFragment>
  <div
    class="w-full max-w-lg mt-2.5 undefined"
  >
    <div
      class="flex justify-between p-2.5"
    >
      <div
        class="text-left"
      >
        <p
          class="text-base font-bold"
        >
          Test Song
        </p>
        <p
          class="text-gray-600 font-medium"
        >
          Test Artist
        </p>
      </div>
      <p
        class="text-gray-400 font-medium text-lg mt-2"
      >
        3:45
      </p>
    </div>
  </div>
</DocumentFragment>
`;
